doc/guides/writing_documentation.rst

```diff
@@ -0,0 +1,104 @@
+:orphan:
+
+=====================
+Writing documentation
+=====================
+
+Types of documentation
+======================
+
+* Tutorials: take a reader unfamiliar with *pydicom* through a series of steps
+  to achieve something useful
+* How-to/examples: more advanced versions of tutorials, for readers that
+  already have some understanding of how *pydicom* works
+* Guides: aim to explain a subject at a fairly high level
+* Reference: contain technical reference information for the *pydicom* API for
+  a reader that has some familiarity with *pydicom* but needs to learn or be
+  reminded about a specific part of it
+
+General style guidelines
+========================
+
+* **pydicom** - italicized lowercase: *pydicom*
+* **DICOM**, **DICOM Standard** - uppercase DICOM, and S on Standard
+* **Python** - capitalize Python
```

Comment:
I'm interested in comments on these style guidelines (particularly *pydicom*). It's what I've been using but am happy to change if we get agreement.